aws/resource_aws_wafv2_ip_set.go

```diff
@@ -0,0 +1,277 @@
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsWafv2IPSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafv2IPSetCreate,
+		Read:   resourceAwsWafv2IPSetRead,
+		Update: resourceAwsWafv2IPSetUpdate,
+		Delete: resourceAwsWafv2IPSetDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected ID/NAME/SCOPE", d.Id())
+				}
+				id := idParts[0]
+				name := idParts[1]
+				scope := idParts[2]
+				d.SetId(id)
+				d.Set("name", name)
+				d.Set("scope", scope)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"addresses": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 50,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 256),
+			},
+			"ip_address_version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					wafv2.IPAddressVersionIpv4,
+					wafv2.IPAddressVersionIpv6,
+				}, false),
+			},
+			"lock_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"scope": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					wafv2.ScopeCloudfront,
+					wafv2.ScopeRegional,
+				}, false),
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsWafv2IPSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	var resp *wafv2.CreateIPSetOutput
+
+	params := &wafv2.CreateIPSetInput{
+		Addresses:        []*string{},
+		IPAddressVersion: aws.String(d.Get("ip_address_version").(string)),
+		Name:             aws.String(d.Get("name").(string)),
+		Scope:            aws.String(d.Get("scope").(string)),
+	}
+
+	if v, ok := d.GetOk("addresses"); ok && v.(*schema.Set).Len() > 0 {
+		params.Addresses = expandStringSet(d.Get("addresses").(*schema.Set))
+	}
+
+	if d.HasChange("description") {
+		params.Description = aws.String(d.Get("description").(string))
+	}
+
+	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
+		params.Tags = keyvaluetags.New(v).IgnoreAws().Wafv2Tags()
+	}
+
+	err := resource.Retry(15*time.Minute, func() *resource.RetryError {
```

Comment:
in my testing, I found these retries can be omitted across the different CRUD operations but if still needed could we move this usage to `config.go`? will make for better re-usability in this resource as well as future ones to come :)

Reply:
I added these retries based on the documentation: https://docs.aws.amazon.com/waf/latest/APIReference/API_CreateIPSet.html
The retry func is added to `config.go`